src/ptvsd/adapter/debuggee.py

```diff
@@ -461,3 +457,45 @@ def _pid_to_handle(pid):
         if not handle:
             raise ctypes.WinError()
         return handle
+
+
+class ProcessPidServer(object):
+ 
+    def __init__(self):
+        self.listener = socket.create_server("127.0.0.1", 0)
+        self.host, self.port = self.listener.getsockname()
+        self.wait_thread = threading.Thread(
+            target=self._wait_for_pid, name="Process Pid Server"
+        )
+        self.wait_thread.daemon = True
+        self.wait_thread.start()
+
+    def _wait_for_pid(self):
```

Comment:
The name is a bit confusing between this and the one without underscore. Since this does all the work to obtain the pid, and it's also inside a class for which this is the sole purpose, I think something like `_worker` would be clearer.